src/webgpu/util/math.ts

```diff
@@ -1128,6 +1128,14 @@ export function cartesianProduct<T>(...inputs: T[][]): T[][] {
  * Recursively calculates all of the permutations, does not cull duplicate
  * entries.
  *
+ * Only feasible for inputs of lengths 5 or so, since the number of permutations
+ * is (inpyt.length)!, so will cause the stack to explode for longer inputs.
+ *
+ * This code could be made iterative using something like
+ * Steinhaus–Johnson–Trotter and additionally turned into a generator to reduce
```

Comment:
TIL!

Reply:
Also, this comment led me to "change ringing" on wikipedia, and then on my music streaming service. So thanks.  :-)

Reply:
A comment so powerful it nerd snipes the reviewer :+1: